src/unique_arc.rs

```diff
@@ -94,9 +94,43 @@ impl<T: ?Sized> UniqueArc<T> {
         debug_assert_eq!(Arc::count(&arc), 1);
         Self(arc)
     }
+
+    /// Creates a new `&mut `[`UniqueArc`] from the given `&mut `[`Arc`].
+    ///
+    /// An unchecked alternative to `Arc::try_as_unique()`
+    ///
+    /// # Safety
+    ///
+    /// The given `Arc` must have a reference count of exactly one
+    pub(crate) unsafe fn from_arc_ref(arc: &mut Arc<T>) -> &mut Self {
+        debug_assert_eq!(Arc::count(&arc), 1);
+
+        // Safety: caller guarantees that `arc` is unique,
+        //         `UniqueArc` is `repr(transparent)`
+        &mut *(arc as *mut Arc<T> as *mut UniqueArc<T>)
+    }
 }
 
 impl<T> UniqueArc<MaybeUninit<T>> {
+    /// Calls `MaybeUninit::write` on the contained value.
+    pub fn write(&mut self, val: T) -> &mut T {
```

Comment:
I'm not sure what convention to use here... Normally arcs/boxes/etc use `this: &mut Self` so there is no conflict between inherent and deref methods. However, calling `MaybeUninit::write` (or `as_mut_ptr`, or `assume_init`) doesn't really make sense, so maybe we *do* want to shadow them? What do you think?

Reply:
You'd just call it UniqueArc::write, no?

Reply:
I mean.. that's possible.

The question is only about "what API is better?". I've left the same as was on `Arc`, but can change it to `this: &mut Self`.